pkg/protobuf/v2/bin/generate-run-protoc

```diff
@@ -2,10 +2,10 @@
 set -euo pipefail
 
 protoc_path=$1
-dir=$(dirname "${protoc_path}")
-script_name="run-protoc"
+script_path=$2
+dir=$(dirname "${script_path}")
```

Comment:
shouldn't this still be 
```suggestion
dir=$(dirname "${protoc_path}")
```
since you're targeting the `protoc_path` not `script_path` ?

Reply:
eg. `generate-run-protoc "/projects/test/artifacts/protobuf/bin/protoc" "artifacts/protobuf/bin/run-protoc"` would internally need to run

`PATH="/projects/test/artifacts/protobuf/bin:${PATH}" "/projects/test/artifacts/protobuf/bin/protoc" "${@}"` to match the protoc plugins path (say that three times fast)

Reply:
No, because with this change, `protoc_path` is no longer guaranteed to be inside the `artifacts` directory. It's the path to whatever `protoc` binary is used on the system.

I've renamed the variables to clarify what each one is. But the second one (that was called `script_path`) is the one you want to take the dirname from, because it's always `artifacts/protobuf/bin/run-protoc`, and hence the dirname ends up as `artifacts/protobuf/bin`. This is where the plugin binaries (like `protoc-gen-go`) that get installed by the makefiles are located.

If you took the dirname of `protoc_path`, you would add the parent directory of whatever `protoc` binary is used on that system to `PATH`. Since the plugins don't get installed there, that doesn't make sense.

Reply:
That makes sense, then it all looks good to me :)